apps/gdalmdimtranslate_lib.cpp

```diff
@@ -1188,29 +1192,65 @@ static bool TranslateArray(
             }
             else
             {
-                GDALGeoTransform gt;
-                if (poSrcDS->GetGeoTransform(gt) == CE_None && gt[2] == 0.0 &&
-                    gt[4] == 0.0)
+                bool bIndexingVarCreated = false;
+                if (srcIndexVar->GetName() == "X" ||
+                    srcIndexVar->GetName() == "Y")
                 {
-                    auto var = std::dynamic_pointer_cast<VRTMDArray>(
-                        poDstGroup->CreateMDArray(
-                            newDimName, {dstDim},
-                            GDALExtendedDataType::Create(GDT_Float64)));
-                    if (var)
+                    GDALGeoTransform gt;
+                    if (poSrcDS->GetGeoTransform(gt) == CE_None &&
+                        gt[2] == 0.0 && gt[4] == 0.0)
                     {
-                        const double dfStart =
-                            srcIndexVar->GetName() == "X"
-                                ? gt[0] + (range.m_nStartIdx + 0.5) * gt[1]
-                                : gt[3] + (range.m_nStartIdx + 0.5) * gt[5];
-                        const double dfIncr =
-                            (srcIndexVar->GetName() == "X" ? gt[1] : gt[5]) *
-                            range.m_nIncr;
-                        std::unique_ptr<VRTMDArraySourceRegularlySpaced>
-                            poSource(new VRTMDArraySourceRegularlySpaced(
-                                dfStart, dfIncr));
-                        var->AddSource(std::move(poSource));
+                        auto var = std::dynamic_pointer_cast<VRTMDArray>(
+                            poDstGroup->CreateMDArray(
+                                newDimName, {dstDim},
+                                GDALExtendedDataType::Create(GDT_Float64)));
+                        if (var)
+                        {
+                            const double dfStart =
+                                srcIndexVar->GetName() == "X"
+                                    ? gt[0] + (range.m_nStartIdx + 0.5) * gt[1]
+                                    : gt[3] + (range.m_nStartIdx + 0.5) * gt[5];
+                            const double dfIncr =
+                                (srcIndexVar->GetName() == "X" ? gt[1]
+                                                               : gt[5]) *
+                                range.m_nIncr;
+                            std::unique_ptr<VRTMDArraySourceRegularlySpaced>
+                                poSource(new VRTMDArraySourceRegularlySpaced(
+                                    dfStart, dfIncr));
+                            var->AddSource(std::move(poSource));
+                            bIndexingVarCreated = true;
+                        }
```

Comment:
Isn't it necessary to log an error in case `var` is null?

Reply:
CreateMDArray() error code paths emit errors